src/notification/notification.js

```diff
@@ -172,6 +180,23 @@ angular.module('patternfly.notification').provider('Notifications', function ()
       }
     };
 
+    notifications.remove = function (notification) {
+      notifications.removeIndex($rootScope.notifications.data.indexOf(notification));
+    };
+
+    notifications.removeIndex = function (index) {
+      $rootScope.notifications.remove(index);
+    };
+
+    notifications.setViewing = function (notification, viewing) {
+      notification.viewing = viewing;
+      if (!viewing && !notification.show) {
+        notifications.remove(notification);
+      }
+    };
+
+    notifications.data = $rootScope.notifications.data;
```

Comment:
The use of $rootScope is from prior to this PR. Since there are (or at least could be) applications built using the Notifications service prior to this, I didn't want to remove it and break them or cause side effects. I didn't much care for its use either so I made it such that applications do not have to use it.